File: dotnet/Allmark/Render/EmphasisRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class EmphasisRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "emphasis",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        RenderTag.Execute(node, state, "em");
    }
}

File: zig/src/render/emphasisRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const renderTagFn = @import("renderTag.zig").renderTag;

pub fn render(node: *const MarkdownNode, state: *RendererState, decode: ?bool) void {
    _ = decode;
    renderTagFn(node, state, "em", true);
}

pub const emphasisRenderer = Renderer{
    .name = "emphasis",
    .render = render,
};
